### DISCUSSION TITLE
How to set the TaxonomyContentItemId using migrations

### DISCUSSION THREAD

**Discussion by MikeAlhayek** [OrchardCMS/OrchardCore#10196] [Q&A]
I am trying to create a content-type called `Inventory` which is constructed using `TitlePart` and `InventoryPart`. The `InventoryPart` will have multiple `TaxonomyField` each will reference a different Taxonomy like (Maker, Model, Color, Fuel type, Body Style, etc)

I am trying to figure out how to correctly use migration to create a Taxonomy field and associate it with taxonomy-content-item. Unless I am mistaken, the property `TaxonomyContentItemId` in the `TaxonomyFieldSettings` needs to be set to the `TaxonomyFieldSettings`. Is this the way to get it done or not?

If so, how can I use the migration to create `TaxonomyContentItem` using a migration to get the id? Also, how can I add the initial list of taxonomy terms to the item using the migration? Perhaps, migrations is not the way to add the `TaxonomyContentItem` and the initial list of terms? Is deployment steps more appropriate for this? Is so, how can I ensure that deployment steps run before the migration? 

Here is what is my migration so far.

```
public int Create()
{
    //create the inventoryPart
    _contentDefinitionManager.AlterPartDefinition(nameof(InventoryPart), part => part
        .WithField(nameof(InventoryPart.Year), field => field
            .OfType(nameof(NumericField))
            .WithDisplayName("Year")
            .MergeSettings<NumericFieldSettings>(settings =>
            {
                settings.Required = true;
                settings.Maximum = 1884;
                settings.Scale = 1;
            }))
        .WithField(nameof(InventoryPart.MakeId), field => field
            .OfType(nameof(TaxonomyField))
            .WithDisplayName("Maker")
            .MergeSettings<TaxonomyFieldSettings>(settings =>
            {
                settings.Required = true,
                settings.Open = true,
                settings.Unique = true,
                //settings.TaxonomyContentItemId where would this come from?
            }))
    );


    // create the inventory content type
    _contentDefinitionManager.AlterTypeDefinition("Inventory", type => type
                        .WithPart(nameof(TitlePart))
                        .WithPart(nameof(InventoryPart)));

    return 1;
}
```

**Comment by tropcicstefan** [comment]
The thing I do is write [recipe ](https://docs.orchardcore.net/en/dev/docs/reference/modules/Recipes/#content-step) for that taxonomy and hardcode contentItemId in a const project (module) that has no references so everyone (every other module) can use it.

If site is new, I create taxonomy on dev machine and export and save that recipe by transforming it into [migration class](https://docs.orchardcore.net/en/dev/docs/reference/modules/Recipes/#recipe-migrations). 
If a site is already in production I create blank taxonomy on production and use that contentItemId for a recipe (same as new), because I need contentItemId that is valid and non existing (no other content item has it) in DB.

That's my way, would like to hear from others too...
